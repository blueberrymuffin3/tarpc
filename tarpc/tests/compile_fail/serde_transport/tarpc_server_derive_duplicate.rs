use tarpc::serde::{Deserialize, Serialize};

#[tarpc::service(derive = "Deserialize", derive = "Serialize")]
trait World {
    async fn hello(name: String) -> String;
}

fn main() {}
