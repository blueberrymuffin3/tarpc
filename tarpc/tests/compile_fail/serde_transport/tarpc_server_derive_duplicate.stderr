error[E0119]: conflicting implementations of trait `Serialize` for type `WorldRequest`
 --> tests/compile_fail/serde_transport/tarpc_server_derive_duplicate.rs:3:1
  |
3 | #[tarpc::service(derive = "Deserialize", derive = "Serialize")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-----------^^
  | |                                                 |
  | |                                                 first implementation here
  | conflicting implementation for `WorldRequest`
  |
  = note: this error originates in the derive macro `tarpc::serde::Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Serialize` for type `WorldResponse`
 --> tests/compile_fail/serde_transport/tarpc_server_derive_duplicate.rs:3:1
  |
3 | #[tarpc::service(derive = "Deserialize", derive = "Serialize")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-----------^^
  | |                                                 |
  | |                                                 first implementation here
  | conflicting implementation for `WorldResponse`
  |
  = note: this error originates in the derive macro `tarpc::serde::Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Deserialize<'_>` for type `WorldRequest`
 --> tests/compile_fail/serde_transport/tarpc_server_derive_duplicate.rs:3:1
  |
3 | #[tarpc::service(derive = "Deserialize", derive = "Serialize")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^-------------^^^^^^^^^^^^^^^^^^^^^^^^
  | |                         |
  | |                         first implementation here
  | conflicting implementation for `WorldRequest`
  |
  = note: this error originates in the derive macro `tarpc::serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Deserialize<'_>` for type `WorldResponse`
 --> tests/compile_fail/serde_transport/tarpc_server_derive_duplicate.rs:3:1
  |
3 | #[tarpc::service(derive = "Deserialize", derive = "Serialize")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^-------------^^^^^^^^^^^^^^^^^^^^^^^^
  | |                         |
  | |                         first implementation here
  | conflicting implementation for `WorldResponse`
  |
  = note: this error originates in the derive macro `tarpc::serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
